### ISSUE TITLE
Support junit 5 `@ParameterizedTest @MethodSource` in `UnusedPrivateMember`

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#4685]
## Current Behavior

`stringProvider` below get flagged for `UnusedPrivateMember`

## Desired Behavior of the rule

In Junit 5 `@ParameterizedTest` you can reference a private function via string (reflection). 
Detekt will flag `stringProvider` as unused, even though it is used by Junit 5. 

Feature request: Special case junit 5's `@MethodSource` and don't report this as `UnusedPrivateMember`

From the docs: 
https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource
```kotlin
    @ParameterizedTest
    @MethodSource("stringProvider")
    fun testWithExplicitLocalMethodSource(argument: String) {
        assertNotNull(argument)
    }

    private fun stringProvider(): Stream<String> {
        return Stream.of("apple", "banana")
    }
```

## Context

Intellij handles this special case, would be nice if detekt did as well.

**Comment by marschwar** [comment]
While this would be an interesting addition, detekt tries to be as framework agnostic as possible. Therefore I do not think this will be added to the detekt rule set. You could however create a custom rule and run that instead of the standard rule.

**Comment by schalkms** [comment]
I agree with @marschwar. The same holds true for #4682.

In this case, the **ignoreAnnotated** config setting can be used.
https://detekt.dev/suppressors.html

**Comment by marschwar** [comment]
> In this case, the ignoreAnnotated config setting can be used.

Without testing this, I don't think that is true as the `@MethodSource` annotation is not on the function that is violating the rule.

**Comment by schalkms** [comment]
@marschwar yeah. You are right. I read over this issue too quickly. I think adding a suppression is fine here.

**Comment by BraisGabin** [comment]
Closing as this is framework specific. If someone came up with a generic way to fix this, please comment.

**Comment by schalkms** [comment]
By the way, IntelliJ 2021.3 correctly detects the function `stringProvider` in the given example above as **used**.
